src/js/base/post-load-hooks.js

```diff
@@ -0,0 +1,136 @@
+          strings_only : runtime.getField(ds, "strings-only"),
+          numbers_only : runtime.getField(ds, "numbers-only"),
+          booleans_only : runtime.getField(ds, "booleans-only"),
+          empty_only : runtime.getField(ds, "empty-only")
+        };
+
+        runtime["makeCStr"] = runtime.getField(ds, "c-str").app;
+        runtime["makeCNum"] = runtime.getField(ds, "c-num").app;
+        runtime["makeCBool"] = runtime.getField(ds, "c-bool").app;
+        runtime["makeCCustom"] = runtime.getField(ds, "c-custom").app;
+        runtime["makeCEmpty"] = function() { return runtime.getField(ds, "c-empty"); };
+
+        runtime["isCStr"] = function(v) { return runtime.unwrap(runtime.getField(ds, "is-c-str").app(v)); };
+        runtime["isCNum"] = function(v) { return runtime.unwrap(runtime.getField(ds, "is-c-num").app(v)); };
+        runtime["isCBool"] = function(v) { return runtime.unwrap(runtime.getField(ds, "is-c-bool").app(v)); };
+        runtime["isCCustom"] = function(v) { return runtime.unwrap(runtime.getField(ds, "is-c-custom").app(v)); };
+        runtime["isCEmpty"] = function(v) { return runtime.unwrap(runtime.getField(ds, "is-c-empty").app(v)); };
+
+        runtime["unwrapCellContent"] = function(v) {
+          if (runtime.isCStr(v)) {
+            return {type: "str", value: runtime.getField(v, "s")};
+          } else if (runtime.isCNum(v)) {
+            return {type: "num", value: runtime.getField(v, "n")};
+          } else if (runtime.isCBool(v)) {
+            return {type: "bool", value: runtime.getField(v, "b")};
+          } else if (runtime.isCCustom(v)) {
+            return {type: "custom", value: runtime.getField(v, "datum")};
+          } else if (runtime.isCEmpty(v)) {
+            return {type: "empty"};
+          } else {
+            runtime.ffi.throwMessageException("Internal error: Cannot unwrap non-cell content");
+          }
+        };
+
+        runtime["makeLoadedTable"] = function(headers, contents) {
+          // Headers can either be [name, sanitizer] arrays or
+          // {name: name, sanitizer: sanitizer} objects
+          headers = headers.map(function(h) {
+            if (h.sanitizer) {
+              return runtime.makeTuple([h.name, h.sanitizer]);
+            } else {
+              return runtime.makeTuple(h);
+            }
+          });
+          return runtime.makeTuple([headers, contents]);
+        };
+        runtime["checkCellContent"] = runtime.makeCheckType(
+          runtime.getField(ds, "is-CellContent").app, "CellContent");
+      },
+      "builtin://reactors": function(reactor) {
+        var r = runtime.getField(runtime.getField(reactor, "provide-plus-types"), "values");
+        runtime.setParam("makeReactor", runtime.getField(r, "make-reactor").app);
+      },
+      "builtin://checker": function(checker) {
+        checker = runtime.getField(runtime.getField(checker, "provide-plus-types"), "values");
```

Comment:
Does this `checker` variable have a declaration? It looks like this is a variable _assignment_ in handalone and a variable _declaration_ in load-lib.

Reply:
I'm not sure, because I can't tell whether it was even _used_ as a variable or not in the original code!  I think it was a copypasta error that was meant to be a variable declaration.

Reply:
I see no `var` declaration for it. Hooray for refactoring fixing bugs!